services/harmony/app/util/cmr.ts

```diff
@@ -749,15 +773,183 @@ async function queryCollections(
  * @param token - Access token for user request
  * @returns The umm collection search results
  */
-async function queryUmmCollections(
+async function _queryUmmCollections(
   context: RequestContext, query: CmrQuery, token: string,
 ): Promise<Array<CmrUmmCollection>> {
+  logger.debug('Calling CMR to fetch UMM collections');
   const ummResponse = await _cmrGet(context, '/search/collections.umm_json_v1_17_3', query, token) as CmrUmmCollectionsResponse;
   return ummResponse.data.items;
 }
 
 /**
- * Performs a CMR grids.umm_json search with the given query string
+ * Generates a consistent MD5 hash for a given CMR query object and token.
+ *
+ * This function:
+ * - Sorts the keys of the query object to ensure consistent ordering.
```

Comment:
sorting the query keys is a good idea - not sure I would have thought to do that